src/main/java/com/example/auctionapp/customValidator/BidVerifierValidator.java

```diff
@@ -0,0 +1,64 @@
+package com.example.auctionapp.customValidator;
+
+import com.example.auctionapp.Util.TimeUtility;
+import com.example.auctionapp.exception.BadRequestException;
+import com.example.auctionapp.model.Bid;
+import com.example.auctionapp.model.Product;
+import com.example.auctionapp.repository.ProductRepository;
+import org.springframework.beans.BeanWrapperImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.List;
+
+public class BidVerifierValidator implements ConstraintValidator<BidVerifier, Object> {
+
+    String productId;
+    String bidTime;
+    String bidAmount;
+    private final ProductRepository productRepository;
+
+    @Autowired
+    public BidVerifierValidator(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    public void initialize(BidVerifier constraintAnnotation) {
+        this.productId = constraintAnnotation.productId();
+        this.bidTime = constraintAnnotation.bidTime();
+        this.bidAmount = constraintAnnotation.bidAmount();
+    }
+
+    public boolean isValid(Object value,
```

Comment:
the validator can be configured to accept the actual class instance, no need for it to be generic `Object`